manager/ui/war/plugins/api-manager/html/policyForms/ip-list.include

```diff
@@ -1,33 +1,48 @@
+    <ul>
+      <li apiman-i18n-key="ip-list-message-example-1">Literal addresses, such as: <code>192.0.2.0</code> or <code>2001:db8::1</code></li>
+      <li apiman-i18n-key="ip-list-message-example-2">CIDR address ranges, such as: <code>192.0.2.0/24</code> or <code>2001:db8:1212::/48</code></li>
+      <li apiman-i18n-key="ip-list-message-example-3">Dashed address ranges, such as: <code>192.0.2.0-192.0.2.10</code>
+      <li apiman-i18n-key="ip-list-message-example-4">Wildcards, such as: <code>192.0.2.*</code></li>
+    </ul>
+  </section>
   <div style="width: 100%; float: left; margin-bottom: 5px; margin-top: 5px">
-    <select id="ip-addresses" ng-model="selectedIP" data-field="ipAddresses" multiple class="apiman-form-control form-control" style="height: 150px; width: 200px; float: left;" ng-options="item for item in config.ipList | orderBy: 'toString()'" ng-disabled="isEntityDisabled()">
+    <select id="ip-addresses" ng-model="selectedIP" data-field="ipAddresses" multiple class="apiman-form-control form-control" style="height: 150px; width: 300px; float: left;" ng-disabled="isEntityDisabled()">
+      <option ng-repeat="item in config.ipList | orderBy: 'toString()'" value="{{item}}">
+        {{item}}
+      </option>
     </select>
     <div style="margin-left: 5px; float: left">
       <button id="clear" ng-click="clear()" ng-disabled="!config.ipList || isEntityDisabled()" data-field="clear" apiman-i18n-key="clear" class="btn btn-default" style="min-width: 75px">Clear</button>
       <div class="clear:both"></div>
       <button id="remove" ng-click="remove(selectedIP)" ng-disabled="!selectedIP || isEntityDisabled()" data-field="remove" apiman-i18n-key="remove" class="btn btn-default" style="min-width: 75px; margin-top: 5px;">Remove</button>
     </div>
   </div>
-  <input id="ip-address" ng-model="ipAddress" ng-pattern="/((^\s*(((?:\*|([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]))\.){3}(?:\*|([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])))\s*$))/" data-field="ipAddress" class="form-control" style="width: 200px; float: left; margin-right: 5px" type="text" apiman-i18n-key="iplist.enter-ip-address" placeholder="Enter an IP address..." ng-disabled="isEntityDisabled()">
-  <button id="add" ng-disabled="!ipAddress || isEntityDisabled()" ng-click="add(ipAddress)" data-field="add" apiman-i18n-key="add" class="btn btn-default" style="min-width: 75px">Add</button>
-</div>
+  <!-- Use ng-model-options allowInvalid, otherwise value is not set until it passes validation, meaning we can't display any useful hints/prompts for user -->
+  <input id="ip-address" name="ipinput" ng-model="ipAddress" data-field="ipAddress" class="form-control" style="max-width: 300px; float: left; margin-right: 5px" type="text" apiman-i18n-key="iplist.enter-ip-address" placeholder="Enter an IP address..." ng-disabled="isEntityDisabled()" ng-model-options="{ updateOn: 'default', allowInvalid:'true' }" validate-ip-address/>
```

Comment:
@msavy Can we try to avoid these inline styles, as they are really annoying to overwrite if someone wants to do a little bit of customization

Reply:
I'm going to completely re-format the TS and HTML of the project once we've done 3.x (using prettier or similar). 

I agree having very long lines like this makes it difficult to spot the difference (I only added `name` and changed width slightly.